nova_vm/src/ecmascript/builtins/indexed_collections/array_objects/array_prototype.rs

```diff
@@ -573,8 +575,18 @@ impl ArrayPrototype {
         Ok(o.into_value())
     }
 
-    fn entries(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    fn entries(agent: &mut Agent, this_value: Value, _: ArgumentsList) -> JsResult<Value> {
+        // 1. Let O be ? ToObject(this value).
+        let Value::Array(o) = this_value else {
+            return Err(
+                agent.throw_exception_with_static_message(ExceptionType::TypeError, "blargh")
```

Comment:
**praise:** blargh indeed

Reply:
I also didn't realize that I changed the implementation of `next` to allow non-array this values but didn't change the array prototype methods to allow arbitrary objects.